### PR TITLE
Add support for metada in createService method

### LINKED ISSUE #52: Allow *.createService to specify metadata
The `createService` verb on Deployment, etc should have the ability to set metadata on the created Service.

### PR DESCRIPTION
**Pull request by Oyelowo**
## Problem
`createService` method on Deployment class uses deployment name with randomised suffix by default and has no means of defining custom name or namespace for a service. Of course, this can be done with the `kx.Service` class but it would be nice to provide overriding options in `createService` method.

This also uses deployment metadata name as service name by default rather than the deployment resource id which provides some stability.

This closes https://github.com/pulumi/pulumi-kubernetesx/issues/52 and https://github.com/pulumi/pulumi-kubernetesx/issues/73

### CODE REVIEW

File: nodejs/kubernetesx/kx.ts [hunk 1]
```diff
@@ -278,16 +282,19 @@ export class Deployment extends k8s.apps.v1.Deployment {
...
+                    type: spec && spec.type as string,
                 };
             });
 
-        return new Service(this.name, {
-            metadata: { namespace: this.metadata.namespace },
+        return new Service(String(metadata?.name) ?? this.name, {
+            metadata: {
+                namespace: this.metadata.namespace,
+                name: metadata?.name ?? this.metadata.name,
```
Comment (danielrbradley) [inline nodejs/kubernetesx/kx.ts:296]: By setting the name here with a default, how will this impact existing users of the library?

**Review by danielrbradley — COMMENTED** [review]
Thanks for the contribution!

I've got 1 question inline as I'm not too familiar with the inner workings of this library. I'll see if I can find someone to give this a second review internally too.

**Comment by danielrbradley** [comment]
Following an internal conversation, we might not be able to release this immeidately as this component isn't actively supported - meaning some of the release mechanism requires additional updates which we've not got capacity to replace. But if you're happy to leave this open then we'll endevour to incorporate this change at the point we can invest more into this component.

Thanks again for the contribution here, sorry we can't get this shipped out as fast as we'd like.

**Comment by mjeffryes** [comment]
Due to lack of tests in this repo and the age, we're not in a good position to move this PR forward so closing for now.

**Pull request closed without merging**